src/btree/bt_random.c

```diff
@@ -394,6 +428,8 @@ __wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, uint32_t flags, WT_
         /* Eviction just wants any random child. */
         if (eviction) {
             descent = pindex->index[__wt_random(rnd) % entries];
+            if (search_inmem_page && __wt_ref_is_root(current))
```

Comment:
Should this be like this?
if (search_inmem_page && __wt_ref_is_root(current)) {
       WT_RET(__random_root_inmem_ref(session, current, &descent, rnd));
} else {
       descent = pindex->index[__wt_random(rnd) % entries];
}

Reply:
What about?
if (search_inmem_page && __wt_ref_is_root(current)) {
WT_RET(__random_root_inmem_ref(session, current, &descent, rnd));
}

if (descent == NULL){
descent = pindex->index[__wt_random(rnd) % entries];
}

Reply:
Then should we also check the page state of descent before doing __random_root_inmem_ref?

Reply:
Not quite, if we dont find anything in __random_root_inmem_ref, this indicated nothing is in mem, so the chosen page is likely on disk and we leave the rest of the code to function accordingly.